mlflow/tracking/utils.py

```diff
@@ -198,11 +182,60 @@ def get_db_profile_from_uri(uri):
     return None
 
 
-def _get_databricks_rest_store(store_uri):
+def _get_databricks_rest_store(store_uri, artifact_uri):  # pylint: disable=W0613
     profile = get_db_profile_from_uri(store_uri)
     return RestStore(lambda: get_databricks_host_creds(profile))
 
 
+class TrackingStoreRegistry:
+
+    def __init__(self):
+        self._registry = {}
+
+    def register(self, scheme, store_builder):
+        self._registry[scheme] = store_builder
+
+    def register_entrypoints(self):
+        # Register tracking stores provided by other packages
+        for entrypoint in entrypoints.get_group_all("mlflow.tracking_store"):
+            self.register(entrypoint.name, entrypoint.load())
+
+    def get_store(self, store_uri=None, artifact_uri=None):
+        store_uri = store_uri if store_uri is not None else get_tracking_uri()
+
+        if store_uri == 'databricks':
+            # Add colon so databricks is parsed as scheme
+            store_uri += ':'
+
+        scheme = urllib.parse.urlparse(store_uri).scheme
+        try:
+            store_builder = self._registry[scheme]
+        except KeyError:
+            raise MlflowException(
+                "Could not find a registered tracking store for: {}".format(store_uri)
+            )
+        return store_builder(store_uri=store_uri, artifact_uri=artifact_uri)
+
+
+_tracking_store_registry = TrackingStoreRegistry()
+_tracking_store_registry.register('', _get_file_store)
+_tracking_store_registry.register('file', _get_file_store)
+_tracking_store_registry.register('databricks', _get_databricks_rest_store)
+
+for scheme in ['http', 'https']:
+    _tracking_store_registry.register(scheme, _get_rest_store)
+
+for scheme in DATABASE_ENGINES:
+    _tracking_store_registry.register(scheme, _get_sqlalchemy_store)
+
+_tracking_store_registry.register_entrypoints()
+
+
+def _get_store(store_uri=None, artifact_uri=None):
```

Comment:
As far as I can tell, `artifact_uri` is only provided by `_download_artifact_from_uri` in order to call `ArtifactRepository.from_artifact_uri`. This is something that may be good to remove in a follow-up PR, assuming we can remove `store` from the arguments of `ArtifactRepostiory`.